src/MessageReader.ts

```diff
@@ -73,7 +75,7 @@ export class MessageReader<T = unknown> {
   ): Record<string, unknown> {
     const msg: Record<string, unknown> = {};
 
-    if (definition.length === 0) {
+    if (!messageDefinitionHasData(definition)) {
```

Comment:
Maybe `messageDefinitionHasDataFields` ? I would expect every definition to have some data unless it's completely empty.